src/main/java/org/jenkinsci/plugins/gitlab/GitlabMergeRequestWrapper.java

```diff
@@ -349,6 +350,22 @@ public GitlabNote createNote(String message, boolean shouldClose, boolean should
 
     }
 
+    public GitlabCommitStatus createCommitStatus(String status, String targetUrl) {
+        try {
+            GitlabAPI api = builder.getGitlab().get();
+            GitlabMergeRequest mergeRequest = api.getMergeRequest(project, id);
+            GitlabCommit latestCommit = getLatestCommit(mergeRequest, api);
+
+            if (latestCommit != null) {
+                return api.createCommitStatus(project, latestCommit.getId(), status, mergeRequest.getSourceBranch(), "Jenkins", targetUrl, null);
```

Comment:
Generally my goal has been for Jenkins to track as little state as possible and rely on GitLab to provide all of the necessary state information. Gitlab Merge Requests point to a particular commit/part of the tree, so it should be possible to expose the necessary fields (if they aren't already exposed in the Java Gitlab API client library). Additionally, the Git plugin and/or Jenkins via SCM should be able to identify what is the current commit that is being built.

That said, if its necessary to track the commit id that the build was started on, then the `GitlabBuildTrigger` should be able to track that state.